kratos/sources/model_part.cpp

```
@@ -855,6 +855,196 @@ void ModelPart::RemoveElementsFromAllLevels(Flags identifier_flag)
 }
 
 
+/* 
+    Functions for Master-Slave Constraint
+*/
+
+/** Inserts a master-slave constraint in the current mesh.
+ */
+void ModelPart::AddMasterSlaveConstraint(ModelPart::MasterSlaveConstraintType::Pointer pNewMasterSlaveConstraint)
+{
+    if (IsSubModelPart())
+    {
+        mpParentModelPart->AddMasterSlaveConstraint(pNewMasterSlaveConstraint);
+    }
+    else
+    {
+        auto existing_condition_it = MasterSlaveConstraints().find(pNewMasterSlaveConstraint->Id());
+        if( existing_condition_it == MasterSlaveConstraintsEnd()) //master-slave constraint did not exist
+        {
+            mMasterSlaveConstraints.insert(mMasterSlaveConstraints.begin(), pNewMasterSlaveConstraint);
+        }
+        else //master-slave constraint did exist already
+        {
+            // In this case, we should update the equation with possible additional masters in the new incoming constraint. 
+            if(&(*existing_condition_it) != (pNewMasterSlaveConstraint.get()))//check if the pointee coincides
```

nope, in this case you should throw an error...